rmw_connextdds_common/src/common/rmw_impl.cpp

```diff
@@ -1540,6 +1557,256 @@ RMW_Connext_Subscriber::take_serialized(
   return rc;
 }
 
+rmw_ret_t
+RMW_Connext_Subscriber::set_cft_expression_parameters(
+  const char * filter_expression,
+  const rcutils_string_array_t * expression_parameters)
+{
+  std::lock_guard<std::mutex> lock(this->cft_mutex);
+  if (nullptr == this->dds_topic_cft) {
```

Comment:
I'll keep this branch, otherwise, the [following logic](https://github.com/ros2/rmw_connextdds/blob/bd36713d9f32d9e9396e48a655981bd2c6cae3e4/rmw_connextdds_common/src/common/rmw_impl.cpp#L1601-L1660) need to add such thing,

```
if ((nullptr == this->dds_topic_cft && has_filter_expression) ||
  ((nullptr != this->dds_topic_cft && !has_filter_expression)))
```

Reply:
I see, sorry, I somehow ignored the `return`, my bad

Reply:
still, it feels a little weird to me... I think I would prefer:

```cpp
if (nullptr == this->dds_topic_cft && filter_expression_empty) {
  ...
  return RMW_RET_OK;
} else if if (nullptr != this->dds_topic_cft) {
  ...
}
```

Reply:
updated